File: sdk/nodejs/src/resources/services.js
"use strict";

const { BaseResource } = require("./base");

class ServicesResource extends BaseResource {
  async list(params = {}) {
    return this._http.get("$/api/service");
  }

  async get(serviceId) {
    return this._http.get("$/api/service/{id}");
  }

  async getMethods(serviceId) {
    return this._http.get("$/api/service/{id}/methods");
  }

  async cancel(serviceId, data) {
    return this._http.post("$/api/service/{id}/cancel", data);
  }

  async getLabel(serviceId) {
    return this._http.get("$/api/service/{id}/label");
  }

  async setLabel(serviceId, data) {
    return this._http.post("$/api/service/{id}/label", data);
  }

  async renew(serviceId) {
    return this._http.post("$/api/service/{id}/renew");
  }

  async getCycle(serviceId) {
    return this._http.get("$/api/service/{id}/cycle");
  }

  async setCycle(serviceId, data) {
    return this._http.post("$/api/service/{id}/cycle", data);
  }

}

module.exports = { ServicesResource };

File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/Services.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class Services extends BaseResource {

    public Services(HttpClient http) {
        super(http);
    }

    /** List services */
    public Map<String, Object> services() throws Exception {
        return http.get("/api/service");
    }

    /** Get service */
    public Map<String, Object> service(int id) throws Exception {
        return http.get(String.format("/api/service/%id", id));
    }

    /** Get service methods */
    public Map<String, Object> methods(int id) throws Exception {
        return http.get(String.format("/api/service/%id/methods", id));
    }

    /** Cancel service */
    public Map<String, Object> cancel(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/cancel", id), data);
    }

    /** Get label */
    public Map<String, Object> label(int id) throws Exception {
        return http.get(String.format("/api/service/%id/label", id));
    }

    /** Set label */
    public Map<String, Object> setLabel(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/label", id), data);
    }

    /** Renew service */
    public Map<String, Object> renew(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/renew", id), data);
    }

    /** Get cycle */
    public Map<String, Object> cycle(int id) throws Exception {
        return http.get(String.format("/api/service/%id/cycle", id));
    }

    /** Set cycle */
    public Map<String, Object> setCycle(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/cycle", id), data);
    }

}
